presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -126,21 +128,20 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
                     .append(switchCaseBlocks);
         }
         else {
-            // TODO: replace Set with fastutils (or similar) primitive sets if types are primitive
-            // for huge IN lists, use a Set
-            Binding constant = generatorContext.getCallSiteBinder().bind(constantValues, Set.class);
+            // for huge IN lists, use a Fastutil primitive Set
+            @SuppressWarnings("unchecked")
+            Set<?> fastutilConstantValuesSet = toFastutilHashSet(constantValues, (Class) javaType);
```

Comment:
You cannot call toFastutilHashSet with javaType as Class<?> because it generates compilation error. constantValues is Object set therefore javaType has to be cast to Class<Object> instead. Even if you use '?' for constantValues set it would still throw an error, because '?' for the set would be different than '?' used in javaType.